### PR TITLE
Add initial QGB Docs

### PR DESCRIPTION
**Pull request by YazzyYaz**
Closes https://github.com/celestiaorg/docs/issues/222

Things Left To Do:

- [x]  Update Mockups
- [x] Show Logs
- [ ] Show EVM Key Generation 
- [ ] Show local QGB Network Instantiation

### CODE REVIEW

File: docs/nodes/qgb.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+## Overview
+
+The Quantum Gravity Bridge (will be referred as QGB for the remainder of this article),
+consists of two components: an Orchestrator and a Relayer.
+
+In the following diagram, we show how an EVM L2 Rollup would post the data to
+Celestia while the Celestia Validators attest to transactions in a one-way bridge
+connected to an EVM L1 Layer (in this case, Ethereum). Then, the EVM L2 Rollup will
+verify the attestations directly on Ethereum on the QGB smart contract. You can
```
Comment (SweeXordious) [inline docs/nodes/qgb.md:19]:
```suggestion
verify the attestations, i.e. valsets and data commitments, directly on Ethereum on the QGB smart contract. You can
```

File: docs/nodes/qgb.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+The following section presumes you have the following setup:
+
+* A Celestia App [Validator](./validator-node.md) running
+* A Celestia Node - [Bridge Node](./bridge-node.md) running
+
+### Validator QGB Flags
+
+Be sure that your Validator node has the following flags running before starting
+the QGB: `--orchestrator-address` and `--evm-address`.
```
Comment (SweeXordious) [inline docs/nodes/qgb.md:58]:
```suggestion
Make sure that your validator has already the correct Orchestrator address and EVM address. These can be specified when creating a new Validator, or editing an existing one, using the following flags: `--orchestrator-address` and `--evm-address`.
```

File: docs/nodes/qgb.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+So, your setup for your Validator node should look like this:
+
+```sh
+celestia-appd start --orchestrator-address <key-name> --evm-address <EVM-ADDRESS>
```
Comment (SweeXordious) [inline docs/nodes/qgb.md:63]:
```suggestion
celestia-appd tx staking create-validator \
    --orchestrator-address="<orch_address>" \
    --evm-address="<evm_address>" \
    ... (other validator creation flags)

File: docs/nodes/qgb.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+```
+
+Here, your `orchestrator-address` can be your Validator address but we recommend
+generating a new Celestia address for the orchestrator.
+
+Your EVM address can be generated with Metamask or any other EVM key generation
+tools in order to get the public address that stars with `0x`.
```
Comment (SweeXordious) [inline docs/nodes/qgb.md:70]:
```suggestion
tools in order to get the public address, in hex format, that stars with `0x`.
```

File: docs/nodes/qgb.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+`keyring-account` refers to the validator key name on Celestia.
+
+Running the orchestrator command above will show logs like the following
+output:
+
+```sh
+SHOW LOGS
+```
+
+With that, you have started the orchestrator! Now, let's move on to the Relayer.
```
Comment (SweeXordious) [inline docs/nodes/qgb.md:100]:
```suggestion
With that, you have started the orchestrator! Now, let's move on to the Relayer.

PS: Validators don't need to run relayers or deploy the QGB smart contract. They can, but it is not a must.
```

**Comment by jcstein** [comment]
image (1)
image (2)
image (3)

File: docs/nodes/qgb.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+## Overview
+
+The Quantum Gravity Bridge (will be referred as QGB for the remainder of this article),
+consists of two components: an Orchestrator and a Relayer.
+
+In the following diagram, we show how a celestium would post the data to
+Celestia while the Celestia Validators attest to transactions in a one-way bridge
+connected to an EVM L1 Layer (in this case, Ethereum). Then, the celestium will
+verify the attestations, i.e. valsets and data commitments, directly on Ethereum
+on the QGB smart contract. You can reference the QGB smart contract [here](https://github.com/celestiaorg/quantum-gravity-bridge/blob/master/src/QuantumGravityBridge.sol).
```
Comment (SweeXordious) [inline docs/nodes/qgb.md:20]:
```suggestion
In the following diagram, we show how a celestium would post the data to
Celestia. This latter which will be attested to by the Celestia validator set, and eventually posted to 
the target EVM chain (in this case, Ethereum). Then, the celestium, or any party, will be able to
verify the attestations, i.e. valsets and data commitments, directly on the EVM chain
on the QGB smart contract. You can reference the QGB smart contract [here](https://github.com/celestiaorg/quantum-gravity-bridge/blob/master/src/QuantumGravityBridge.sol).
```

File: docs/nodes/qgb.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+![QGB-Architecture](/img/nodes/qgb-diagram.png)
+
+The specification of the QGB `Valset`s, which track the Celestia validator set
```
Comment (SweeXordious) [inline docs/nodes/qgb.md:24]:
```suggestion
The specification of the QGB valsets, which track the Celestia validator set
```

File: docs/nodes/qgb.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+  be signed
+* The Orchestrator: queries the attestation, signs it, then submits the signature
+  back to Celestia App
+
+The diagram below goes over this process.
+
+![QGB-Orchestrator](/img/nodes/qgb-orchestrator.png)
+
+The Relayer deploys the QGB smart contract first to the EVM L1 chain (if it is
+not deployed before) and then relays the attestations from Celestia App to the
```
Comment (SweeXordious) [inline docs/nodes/qgb.md:42]:
```suggestion
not deployed before), and then relays the attestations from Celestia App to the
```

**Review by SweeXordious — CHANGES_REQUESTED** [review]
second pass, added more suggestions

**Comment by SweeXordious** [comment]
@YazzyYaz @jcstein does it make sense to add a section where you can run a local version of the QGB? Or this should belong in the repo?

**Comment by jcstein** [comment]
@SweeXordious I think it does make sense to include, but maybe not for first iteration so we can push this live?

**Comment by SweeXordious** [comment]
@jcstein Yes, 100%, sounds good, we can remember it or create an issue for it.

**Comment by YazzyYaz** [comment]
@SweeXordious where can we find examples of running a local version of the QGB?

Is it the init-chain.sh or docker-compose build or both? Happy to add them to the PR.

**Comment by SweeXordious** [comment]
Which do you prefer? A network where they have to run similar to running a validator?
Or an automatic docker compose script that starts with a single command and spins up eveytuing automatically?

File: docs/nodes/qgb.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+The `-d` flag refers to your EVM Private Key that will be used to sign the
+transactions.
+
+`keyring-account` refers to the validator key name on Celestia.
+
+Running the orchestrator command above will show logs like the following
+output:
+
+```sh
+SHOW LOGS
```
Comment (SweeXordious) [inline docs/nodes/qgb.md:106]:
```suggestion
D[2022-11-02|23:04:32.758] initializing orchestrator                    
D[2022-11-02|23:04:32.766] starting orchestrator                        
I[2022-11-02|23:04:32.766] listening for new block events...            
I[2022-11-02|23:04:32.768] syncing missing nonces                       latest_nonce=2 last_unbonding_height=0
D[2022-11-02|23:04:32.768] enqueueing missing attestation nonce         nonce=2
D[2022-11-02|23:04:32.768] enqueueing missing attestation nonce         nonce=1
I[2022-11-02|23:04:32.768] finished syncing missing nonces              latest_nonce=2 last_unbonding_height=0
E[2022-11-02|23:04:32.768] stopping enqueing missing attestations       
D[2022-11-02|23:04:32.768] processing nonce                             nonce=2
I[2022-11-02|23:04:32.919] signed commitment                            nonce=2 begin_block=0 end_block=5 commitment=A288BB22829DFF30508A7D22C3943507D3056D6FDEA3E42D845E55277D991996 tx_hash=BC372BF3018CF57A924E9687942EAC3FAC01688D943E5E62B9C85D9DD80175F4
D[2022-11-02|23:04:32.919] processing nonce                             nonce=1
I[2022-11-02|23:04:33.938] signed Valset                                nonce=1 tx_hash=0C4F6312D138CEA3D65DA4D5084C0A7BE4B8E680D083A326E150EC23DCBB8FC4
D[2022-11-02|23:04:36.993] enqueueing new attestation nonce             nonce=3
D[2022-11-02|23:04:36.993] processing nonce                             nonce=3
I[2022-11-02|23:04:38.012] signed commitment                            nonce=3 begin_block=5 end_block=10 commitment=7132B548311BA737EEC4085096B34FE436DCC5A78801C3E58F2AD20992567006 tx_hash=358702C95CC848D1511934A894AD9784507FB826BFDE31D2685E704111803302
D[2022-11-02|23:04:42.097] enqueueing new attestation nonce             nonce=4
D[2022-11-02|23:04:42.097] processing nonce                             nonce=4
I[2022-11-02|23:04:43.106] signed commitment                            nonce=4 begin_block=10 end_block=15 commitment=D541B8F010C615172570B181BBE57A8C2314CE3B9E785EF831EC2FF0A5B8D93A tx_hash=FBC01DAFBEE82B02F8041574B05154C9C09A8723A06F66B327F13C6D869868EE
```

File: docs/nodes/qgb.md [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+
+The `-e` flag references the RPC endpoint for Sepolia in order to connect to it.
+
+The `-z` flag references the Chain ID of Sepolia.
+
+When you run the command, you should get the following output that indicates
+the QGB Smart Contract has been deployed:
+
+```sh
+SHOW LOGS
```
Comment (SweeXordious) [inline docs/nodes/qgb.md:138]:
```suggestion
I[2022-11-02|23:05:37.293] deployed QGB contract                        address=0x735e796b72c13E427C8eD9d1a24f6e6147217CE1 hash=0xbc06b4b861f6e9ffbd4c3f8bb393085182a441141e906e7b3b4f5cb38584c3cc
```

File: docs/nodes/qgb.md [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+
+The `-a` flag must be the QGB smart contract address that you received in the output
+when you deployed the smart contract in the previous section.
+
+The `-l` flag refers to the gas limit being set for the Relayer.
+
+Running the command will give you the following output:
+
+```sh
+SHOW LOG
```
Comment (SweeXordious) [inline docs/nodes/qgb.md:166]:
```suggestion
D[2022-11-02|23:06:18.029] found enough data commitment confirms to be relayed majThreshHold=2863311532 currThreshold=4294967296
I[2022-11-02|23:06:18.030] relaying data commitment 0-5...              
D[2022-11-02|23:06:18.813] waiting for data commitment to be confirmed  nonce=2 hash=0x16d4762e4955691589a1e8f22cd0f25e9f3a7e6783e2b014f708ddd8005fb58b
I[2022-11-02|23:06:37.018] relayed data commitment                      nonce=2 hash=0x16d4762e4955691589a1e8f22cd0f25e9f3a7e6783e2b014f708ddd8005fb58b
D[2022-11-02|23:06:37.206] found enough data commitment confirms to be relayed majThreshHold=2863311532 currThreshold=4294967296
I[2022-11-02|23:06:37.207] relaying data commitment 5-10...             
D[2022-11-02|23:06:38.023] waiting for data commitment to be confirmed  nonce=3 hash=0xe142d968897c20962211da34e4b61a38379d14e24f0057074bedce6028dc0c76
```

**Review by SweeXordious — COMMENTED** [review]
Added some logs + how to run details

**Comment by jcstein** [comment]
Thank you, I'll review when back home @SweeXordious!

File: docs/nodes/qgb.md [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+You can run it with the following command:
+
+```sh
+cd celestia-app/
+bash script/single-node.sh
+```
+
+This starts a local network with QGB enabled. You can then run the Orchestrator
+and Relayer after which were covered in the previous sections.
+
```
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/nodes/qgb.md:189]: Hmm.. should we replace the private key with <EVM_PRIVATE_KEY> @SweeXordious?

[the remark this answers is not in the document]
Comment (SweeXordious) [reply docs/nodes/qgb.md:189]: That makes sense. But would require updating the single node script, since it has the corresponding address hardcoded. 
So, I guess for now, we can leave it that way.
But I'm fine if you decide to update the single node script, or copy the commands there to this doc and ask the devs to run them one by one.
Whichever you prefer :D

[the remark this answers is not in the document]
Comment (jcstein) [reply docs/nodes/qgb.md:189]: Ahh - okay that makes sense. I'll ask you a little more about this next week 😄 thanks Rachid

File: docs/nodes/qgb.md [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+The `-x` flag references the chain-id of the Celestia network. Here, we are
+connecting it to Mocha Testnet so we are using `mocha`.  
+
+The `-d` flag refers to your EVM Private Key that will be used to sign the
+transactions.
+
+`keyring-account` refers to the validator key name on Celestia.
+
+Running the orchestrator command above will show logs like the following
+output:
```
[the remark this answers is not in the document]
Comment (SweeXordious) [reply docs/nodes/qgb.md:103]: good point :+1: :+1:. This was written before we decided to have a separate testnet for the QGB.
So, we can remove these docs from here, and keep them somewhere until we start the QGB dedicated testnet. Thanks a lot for flagging this.

File: docs/nodes/qgb.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/nodes/qgb.md:214]: i added these suggestions. @YazzyYaz let's figure out the best way to deal with the private key on here though

File: docs/nodes/qgb.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/nodes/qgb.md:104]: hmm, should i leave out lines 81-268 out for V1?

File: yarn.lock [hunk 14]
```diff
@@ -131,17 +131,17 @@
     "@jridgewell/gen-mapping" "^0.1.0"
     "@jridgewell/trace-mapping" "^0.3.9"
 
-"@babel/code-frame@^7.0.0", "@babel/code-frame@^7.10.4", "@babel/code-frame@^7.16.0", "@babel/code-frame@^7.16.7", "@babel/code-frame@^7.8.3":
-  version "7.16.7"
-  resolved "https://registry.npmjs.org/@babel/code-frame/-/code-frame-7.16.7.tgz"
-  integrity sha512-iAXqUn8IIeBTNd72xsFlgaXHkMBMt6y4HJp1tIaK465CWLT/fG1aqB7ykr95gHHmlBdGbFeWWfyB4NJJ0nmeIg==
+"@babel/code-frame@^7.0.0", "@babel/code-frame@^7.10.4", "@babel/code-frame@^7.16.0", "@babel/code-frame@^7.18.6", "@babel/code-frame@^7.8.3":
```
Comment (jcstein) [inline yarn.lock:134]: and I think we can revert the changes here @YazzyYaz?

File: docs/nodes/qgb.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (SweeXordious) [reply docs/nodes/qgb.md:104]: yes :+1: let's keep those somewhere because we will need them in the future.

[the remark this answers is not in the document]
Comment (jcstein) [reply docs/nodes/qgb.md:104]: sounds good. will review with @YazzyYaz

[the remark this answers is not in the document]
Comment (evan-forbes) [reply docs/nodes/qgb.md]: I'll defer to @YazzyYaz , but imo yeah. alternatively, we could just mention that we're not running the qgb on a testnet atm so these docs are not yet relevant. The end result for v2 will likely be similar, and this info is really good

[the remark this answers is not in the document]
Comment (jcstein) [reply docs/nodes/qgb.md]: sounds good, thank you Evan

**Comment by jcstein** [comment]
closing in favor of #668
